uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-notification.component.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-env amd */
+
+define([
+  'leave-absences/shared/modules/components'
+], function (components) {
+  components.component('leaveNotification', {
+    bindings: {
+      eventName: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-notification.html';
+    }],
+    controllerAs: 'notification',
+    controller: ['$log', '$rootScope', 'pubSub', 'LeaveRequest', controller]
+  });
+
+  function controller ($log, $rootScope, pubSub, LeaveRequest) {
+    $log.debug('Component: leave-notification');
+
+    var filters = {};
+    var vm = this;
+    vm.count = 0;
+    vm.loading = { count: true };
+
+    (function init () {
+      initializeListeners();
+    })();
+
+    /**
+     * Fetch count of leave requests which matches the filter
+     *
+     * @return {Promise}
+     */
+    function fetchCount () {
+      vm.loading.count = true;
+
+      return LeaveRequest.all(filters)
+        .then(function (leaveRequests) {
+          vm.count = leaveRequests.list.length;
+          vm.loading.count = false;
+        });
+    }
+
+    /**
+     * Initializes the filter and fetches fetches filtered leave requests
+     * @param {Object} e - Event object
+     * @param {Object} filtersData - Filters
+     * @return {Promise}
+     */
+    function initializeFilters (e, filtersData) {
+      filters = filtersData;
+
+      return fetchCount();
+    }
+
+    /**
+     * Initializes the event listeners
+     */
+    function initializeListeners () {
+      $rootScope.$on('WaitingApproval:: Initialize Filters::' + vm.eventName, initializeFilters);
```

Comment:
I don't think the `<leave-notification-badge />` needs to care how `filter` (or `eventName`, for all that matter) are set in the parent component, all it needs to know is that whoever wants to use it, it should give those two params.

Right now the signature is confusing, cause half is provided by binding, half is provided via event, while it should be something as simple as
```html
<leave-notification-badge filters="theFilters" eventName="theEventName"></leave-notification-badge>
```
if the parent component needs to asynchronously fetch the filters first, then let the parent component handle that, `<leave-notification-badge />` should not care about it

It could be a simply as not injecting the component until `vm.filters` is defined, or something similar

```html
<leave-notification-badge ng-if="$ctrl.theFilters"></leave-notification-badge>
```